File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& mt, int t) {
        int m=mt.size();
        int n=mt[0].size();

        int row=m-1;
        int col=0;

        while(row>=0 && col<n)
        {
            if(mt[row][col]==t)
            {
                return true;
            }

            else if(mt[row][col]<t)
            {
                col++;
            }
            else{
                row--;
            }
            
        }
        return false;
        
    }
};

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] mat, int target) {
        
        int m = mat.length;
        int n = mat[0].length;
        int left = 0 , right = m*n-1;

        while(left<=right)
        {
            int mid = left + (right -left) / 2 ;

            int val = mat[mid/n][mid%n];

            if(val==target) return true;

            else if( val<target) left = mid + 1 ; 

            else right = mid - 1 ;
        }

        return false ;
    }
}